app/components/post_header/index.js

```diff
@@ -23,7 +23,7 @@ function makeMapStateToProps() {
         const militaryTime = getBool(state, Preferences.CATEGORY_DISPLAY_SETTINGS, 'use_military_time');
 
         return {
-            commentedOnDisplayName: displayUsername(commentedOnUser, teammateNameDisplay),
+            commentedOnDisplayName: commentedOnUser ? displayUsername(commentedOnUser, teammateNameDisplay) : '',
```

Comment:
@enahum This does resort to the expected behaviour but the logic still might cause issues based on the `username` logic again as it is dependent on it. 

If expected is to render header `commented on xyz`  on messages with replies and the first message in grouped messages then

i am more inclined with something of the lines dependent on the respective conditions

        if (!this.props.renderReplies || !this.props.isFirstReply) {
            return null;
        } 

instead of https://github.com/mattermost/mattermost-mobile/blob/37a6b3c6d6066fb7ebdf77b9f8b1c3992f0517ae/app/components/post_header/post_header.js#L110